app/rpc/rpccontext/verbosedata.go

```diff
@@ -93,6 +89,10 @@ func (ctx *Context) PopulateBlockWithVerboseData(block *appmessage.RPCBlock, dom
 		}
 	}
 
+	if domainBlock == nil && blockInfo.BlockStatus == externalapi.StatusHeaderOnly {
+		return nil
+	}
+
```

Comment:
This condition is always true, since `GetBlockEvenIfHeaderOnly` returns an error if the block is not found. I guess this is the condition you wanted to check?
```suggestion
	if len(domainBlock.Transactions) == 0 {
		return nil
	}

```